test/combinations.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const getName = (prefix, combi) => {
+  const name = Object.entries(combi)
+    .map(([prop, value]) => {
+      if (typeof value === 'boolean') {
+        return value ? prop : `regular`;
+      }
+      return value;
+    })
+    .join('/');
+
+  return `${prefix}/${name}`;
+};
+
+const computeCombinations = entries => {
```

Comment:
Can you maybe add a few lines of docs what the input and output here is?

Reply:
This is unlikely to be merged right now (as it's also incomplete). We'll discuss whether the currently created ui-kit sketch file is useful to designers. If so, we want to improve the setup and I'll polish it some more :)

There are some use-cases of the `Combinations` component. What it does is

It takes an object of prop-names and their possible values

```js
{
  disabled: [true, false],
  readonly: [true, false]
}
```

and then calls the `render` function once for each combinations.

In our example once for each of these:

```js
[
  { disabled: true, readonly: true },
  { disabled: true, readonly: false },
  { disabled: false, readonly: true },
  { disabled: true, readonly: false },
]
```

The specific functions use intermediate data formats to be able to build up the tree first and then the `flattenRemainingEntries` flattens them down into the actual combinations.

<details>
<summary>Details</summary>

```js
// this is what the config looks like
const config = { disabled: [true, false], readonly: [true, false], horizontalConstraint: ['s', 'm'] }
```

```js
// computeCombinations
// takes entries like [[propName, possibleValues], ..] produced by Object.entries(config)
const entries = [
  ['disabled', [true, false]],
  ['readonly', [true, false]]
]
// and returns
[
  {
    name: 'disabled',
    value: true,
    remainingEntries: [{ name: 'readonly', value: true }, { name: 'readonly', value: false }]
  },
  {
    name: 'disabled',
    value: false,
    remainingEntries: [{ name: 'readonly', value: true }, { name: 'readonly', value: false }]
  },
]
```

```js
// flattenRemainingEntries
// takes
{
  name: 'disabled',
  value: true,
  remainingEntries: [{ name: 'readonly', value: true }, { name: 'readonly', value: false }]
}
// and returns
[
  { disabled: true, readonly: true },
  { disabled: true, readonly: false },
]
```

</details>